ompi/mca/pml/ob1/pml_ob1_progress.c

```diff
@@ -54,8 +56,8 @@ static inline int mca_pml_ob1_process_pending_cuda_async_copies(void)
 static int mca_pml_ob1_progress_needed = 0;
 int mca_pml_ob1_enable_progress(int32_t count)
 {
-    int32_t old = OPAL_ATOMIC_ADD32(&mca_pml_ob1_progress_needed, count);
-    if( 0 != old )
+    int32_t progress_count = OPAL_ATOMIC_ADD32(&mca_pml_ob1_progress_needed, count);
```

Comment:
if mca_pml_ob1_progress_needed is 0 and count is 1, then the current version returns, but the patched one does not
OPAL_ATOMIC_ADD32 returns the __new__ value, and __not__ the old one, so the current version is clearly misleading, and imho, busted too